ArduPlane/tailsitter.cpp

```diff
@@ -845,7 +847,11 @@ void Tailsitter_Transition::update()
         quadplane.assisted_flight = true;
         uint32_t dt = now - fw_transition_start_ms;
         // multiply by 0.1 to convert (degrees/second * milliseconds) to centi degrees
+#if AP_INVERTED_FLIGHT_ENABLED
         plane.nav_pitch_cd = constrain_float(fw_transition_initial_pitch - (quadplane.tailsitter.transition_rate_fw * dt) * 0.1f * (plane.fly_inverted()?-1.0f:1.0f), -8500, 8500);
+#else
+        plane.nav_pitch_cd = constrain_float(fw_transition_initial_pitch - (quadplane.tailsitter.transition_rate_fw * dt) * 0.1f, -8500, 8500);
+#endif
```

Comment:
I would prefer to pull out either the sign or the fly inverted bool and keep just one line of the constrain. I think the maths is easier to understand if there is only one copy.

Reply:
Maybe we could take the opportunity to break it up a little. 
```suggestion
        float pitch_ramp = quadplane.tailsitter.transition_rate_fw * dt * 0.1;
#if AP_INVERTED_FLIGHT_ENABLED
        if (plane.fly_inverted()) {
            // Transitioning to inverted flight, pitch up rather than down
            pitch_ramp *= -1.0;
        }
#endif        
        plane.nav_pitch_cd = constrain_float(fw_transition_initial_pitch - pitch_ramp, -8500, 8500);
```

Reply:
> Maybe we could take the opportunity to break it up a little.

Yep, I did that, and random other changes like it.

And reverted those changes, because I *really* like to keep these add-defines things as completely no compiler output change in the case that you have not turned the option on.  It means that no existing user will ever run into problems with the code.

I've achieved that with this PR now, and updated the blocks in the opening comment appropriately.

It *does* uglify the code a bit, but I *do* plan on going back to the places and cleaning it up by factoring bits and pieces.  Especially the bit in takeoff.cpp.  The block that patch is in will make for a nice new method with earl returns etc, something like:

```
    if (do_takeoff_attitude_check &&
         !takeoff_attitude_is_good()) {
            gcs().send_text(MAV_SEVERITY_WARNING, "Bad launch AUTO");
            takeoff_state.accel_event_counter = 0;
            goto no_launch;
        }
    }
```

```
        // Check aircraft attitude for bad launch
        bool takeoff_attitude_is_good(void) const
        {
            if (ahrs.pitch_sensor <= -3000 || ahrs.pitch_sensor >= 4500) {
                return false;
            }
#if AP_INVERTED_FLIGHT_ENABLED
            if (fly_inverted()) {
                // skip roll check if we're *supposed* to be upside-down
                return true;
            }
#if endif
            if (labs(ahrs.roll_sensor) > 3000) {
                    return false;
            }
            return true;
        }
    }
```

I think that's the same logic, but I'm not convinced it's good logic.... e.g. should those pitch limits be the other way around if you're inverted?  Why aren't we checking roll when inverted?

Give I think we now have a partner who flies this way, we should probably give it more serious consideration now, and I think this would make for a good separate PR...

Reply:
I've created a patch on top of this over here which shows things prettied up a bit:
https://github.com/ArduPilot/ardupilot/compare/master...peterbarker:ardupilot:pr/inverted-flight-tidy?expand=1